src/Components/Server/src/Circuits/ArrayBuilderMemoryStream.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Web.Rendering
+{
+    /// <summary>
+    /// Writeable memory stream backed by a an <see cref="ArrayPool{T}"/>.
+    /// </summary>
+    internal sealed class ArrayBuilderMemoryStream : Stream
+    {
+        public ArrayBuilderMemoryStream()
+        {
+            ArrayBuilder = new ArrayBuilder<byte>(2048);
```

Comment:
This `2048` value seems very coupled to the current use case. Maybe make it a constructor param?

Reply:
Or let the consumer in the `ArrayBuilder`

Reply:
This sounds like a great idea.